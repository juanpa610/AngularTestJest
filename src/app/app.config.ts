import { ApplicationConfig } from '@angular/core';
import { provideRouter } from '@angular/router';

import { routes } from './app.routes';
import { provideHttpClient, withFetch } from '@angular/common/http';
import { AlbumGateway } from './domain/gateway/album.gateway';
import { AlbumService } from './infraestructure/driven-adapters/album/album.service';

export const appConfig: ApplicationConfig = {
  providers: [
    provideRouter(routes),
    provideHttpClient(withFetch()),
    {
      provide: AlbumGateway,
      useClass: AlbumService
    }
  ]
};
