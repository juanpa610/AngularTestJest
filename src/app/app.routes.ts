import { Routes } from '@angular/router';

export const routes: Routes = [

    {
        path: '',
        redirectTo: 'album',
        pathMatch: 'full'
    },
    {
        path: 'album',
        loadChildren: () => import('./ui/albums/albums.module').then(m => m.AlbumsModule)
    }

];
