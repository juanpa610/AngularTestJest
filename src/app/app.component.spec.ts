import { ComponentFixture, TestBed } from '@angular/core/testing';
import { AppComponent } from './app.component';

describe('AppComponent', () => {
  let compoponent: AppComponent;
  let fixture: ComponentFixture<AppComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [AppComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(AppComponent);
    compoponent = fixture.componentInstance;
  });

  it('should create the app', () => {
    expect(compoponent).toBeTruthy();
  });

  it(`should have the 'JestAngular-17-2-0' title`, () => {
    expect(compoponent.title).toEqual('JestAngular-17-2-0');
  });

  it('should render title', () => {
    fixture.detectChanges();
    const compiled = fixture.nativeElement as HTMLElement;
    expect(compiled.querySelector('h1')?.textContent).toContain('Hello, JestAngular-17-2-0');
  });

  it('should do match with snapshot', () => {
    fixture.detectChanges();
    const compiled = fixture.nativeElement as HTMLElement;
    expect(compiled).toMatchSnapshot();
  });
});
