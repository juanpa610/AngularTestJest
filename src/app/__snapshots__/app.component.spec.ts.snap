// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should do match with snapshot 1`] = `
<div
  id="root3"
>
  <h1>
    Hello, JestAngular-17-2-0
  </h1>
  <p>
    Congratulations! Your app is running. ðŸŽ‰
  </p>
  <router-outlet />
  
</div>
`;
